src/libraries/System.Net.Sockets/tests/FunctionalTests/Connect.cs

```diff
@@ -137,6 +137,12 @@ public async Task Connect_AfterDisconnect_Fails()
         [InlineData("[::ffff:1.1.1.1]", true, false)]
         public async Task ConnectGetsCanceledByDispose(string addressString, bool useDns, bool owning)
         {
+            if (UsesSync && !PlatformDetection.IsWindows)
```

Comment:
Is there need to disable this also on macOS? It feels like `!IsWindows` should really be `IsLinux`????
Also this is regression in particular Linux Kernels. At that seems rare at this moment. It may be worth of adding version check instead of blindly disabling Linux runs -> it should still work on most supported distributions AFAIK.

Reply:
Ah yes, I've missed it. It should work as expected on macOS, thanks for catching that. I'll change it with another PR